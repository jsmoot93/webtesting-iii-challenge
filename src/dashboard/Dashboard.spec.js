import React from "react";
import { render, fireEvent, cleanup } from "react-testing-library";

import Dashboard from "./Dashboard";

afterEach(cleanup);

describe("Dashboard Tests", () => {
  it("Should match snapshot", () => {
    const { container } = render(<Dashboard />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("renders successfully without crashing", () => {
    render(<Dashboard />);
  });
  it("should render controls and display", () => {
    render(<Dashboard />);

    expect(document.querySelector(".controls")).toBeTruthy();
    expect(document.querySelector(".display")).toBeTruthy();
  });
});
