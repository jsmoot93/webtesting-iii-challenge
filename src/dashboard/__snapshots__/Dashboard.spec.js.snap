// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Tests Should match snapshot 1`] = `
<div
  class="display panel"
>
  <div
    class="led green-led"
    data-testid="locked-led"
  >
    Unlocked
  </div>
  <div
    class="led green-led"
    data-testid="closed-led"
  >
    Open
  </div>
</div>
`;
